aws/resource_aws_docdb_cluster_parameter_group.go

```diff
@@ -0,0 +1,282 @@
+	tags := tagsFromMapDocDB(d.Get("tags").(map[string]interface{}))
+
+	var groupName string
+	if v, ok := d.GetOk("name"); ok {
+		groupName = v.(string)
+	} else if v, ok := d.GetOk("name_prefix"); ok {
+		groupName = resource.PrefixedUniqueId(v.(string))
+	} else {
+		groupName = resource.UniqueId()
+	}
+
+	createOpts := docdb.CreateDBClusterParameterGroupInput{
+		DBClusterParameterGroupName: aws.String(groupName),
+		DBParameterGroupFamily:      aws.String(d.Get("family").(string)),
+		Description:                 aws.String(d.Get("description").(string)),
+		Tags:                        tags,
+	}
+
+	log.Printf("[DEBUG] Create DocDB Cluster Parameter Group: %#v", createOpts)
+
+	resp, err := conn.CreateDBClusterParameterGroup(&createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating DocDB Cluster Parameter Group: %s", err)
+	}
+
+	d.SetId(aws.StringValue(createOpts.DBClusterParameterGroupName))
+	log.Printf("[INFO] Neptune Cluster Parameter Group ID: %s", d.Id())
+
+	d.Set("arn", resp.DBClusterParameterGroup.DBClusterParameterGroupArn)
+
+	return resourceAwsDocDBClusterParameterGroupUpdate(d, meta)
+}
+
+func resourceAwsDocDBClusterParameterGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).docdbconn
+
+	describeOpts := &docdb.DescribeDBClusterParameterGroupsInput{
+		DBClusterParameterGroupName: aws.String(d.Id()),
+	}
+
+	describeResp, err := conn.DescribeDBClusterParameterGroups(describeOpts)
+	if err != nil {
+		if isAWSErr(err, docdb.ErrCodeDBParameterGroupNotFoundFault, "") {
+			log.Printf("[WARN] DocDB Cluster Parameter Group (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+	}
+
+	if len(describeResp.DBClusterParameterGroups) != 1 ||
+		*describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupName != d.Id() {
+		return fmt.Errorf("Unable to find Cluster Parameter Group: %#v", describeResp.DBClusterParameterGroups)
+	}
+
+	arn := aws.StringValue(describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupArn)
+	d.Set("arn", arn)
+	d.Set("description", describeResp.DBClusterParameterGroups[0].Description)
+	d.Set("family", describeResp.DBClusterParameterGroups[0].DBParameterGroupFamily)
+	d.Set("name", describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupName)
+
+	describeParametersOpts := &docdb.DescribeDBClusterParametersInput{
+		DBClusterParameterGroupName: aws.String(d.Id()),
+		Source:                      aws.String("user"),
+	}
+
+	describeParametersResp, err := conn.DescribeDBClusterParameters(describeParametersOpts)
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("parameter", flattenDocDBParameters(describeParametersResp.Parameters)); err != nil {
+		return fmt.Errorf("error setting docdb cluster parameter: %s", err)
+	}
+
+	resp, err := conn.ListTagsForResource(&docdb.ListTagsForResourceInput{
+		ResourceName: aws.String(arn),
+	})
+
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", arn)
```

Comment:
We should return an error here instead of a debug log:

```suggestion
		return fmt.Errorf("error listing tags for DocDB Cluster Parameter Group (%s): %s", d.Id(), err)
```

Reply:
Fixed